utils/misc.mli

```diff
@@ -682,3 +682,25 @@ module Magic_number : sig
 
   val all_kinds : kind list
 end
+
+module RevList : sig
+  type 'a t
+  (** Lists with an efficient "cons" at the end of the list. *)
+
+  val empty : 'a t
+  val to_list : 'a t -> 'a list
+  val to_list_append : 'a t -> 'a list -> 'a list
```

Comment:
on callsites I would find `append_to_list` more natural, and actually I think that `rev_append` would be fine.